waspc/packages/wasp-config/src/run.ts

```diff
@@ -77,14 +96,14 @@ function parseProcessArguments(args: string[]): {
   }
 }
 
-function parseEntityNamesJson(entitiesJson: string): string[] {
+function getValidEntityNamesOrThrow(entitiesJson: string): string[] {
   const entities = JSON.parse(entitiesJson)
   if (!Array.isArray(entities)) {
     throw new Error('The entities JSON must be an array of entity names.')
   }
   return entities
 }
 
-function serialize(appConfig: Decl[]): string {
-  return JSON.stringify(appConfig)
-}
+type Result<Value, Error> =
+  | { status: 'ok'; value: Value }
+  | { status: 'error'; error: Error }
```

Comment:
I'm using this for normal control flow errors (e.g., user forgets something).

I'm using exceptions for unexpected errors/compiler bugs. For example, an exception is thrown if `waspc` doesn't pass the correct arguments to this executable.

For the normal control flow, I am using the `Result` type instead of exceptions because:
- The error users see is nicer (no stack trace)
- They can slip through the type system.

Reply:
Nice! If only there was a language that already used this kind of approach for errors/exceptions :D!